plugins/services/src/js/components/forms/MultiContainerNetworkingFormSection.tsx

```diff
@@ -26,40 +24,61 @@ import FormRow from "#SRC/js/components/form/FormRow";
 import InfoTooltipIcon from "#SRC/js/components/form/InfoTooltipIcon";
 import MetadataStore from "#SRC/js/stores/MetadataStore";
 import Networking from "#SRC/js/constants/Networking";
-import StoreMixin from "#SRC/js/mixins/StoreMixin";
 import * as ValidatorUtil from "#SRC/js/utils/ValidatorUtil";
-import VirtualNetworksStore from "#SRC/js/stores/VirtualNetworksStore";
 import VipLabelUtil from "../../utils/VipLabelUtil";
 
 import { FormReducer as networks } from "../../reducers/serviceForm/MultiContainerNetwork";
 import ServiceConfigUtil from "../../utils/ServiceConfigUtil";
 import { getHostPortPlaceholder, isHostNetwork } from "../../utils/NetworkUtil";
+import VirtualNetworksActions from "#SRC/js/events/VirtualNetworksActions";
+import { Overlay } from "#SRC/js/structs/Overlay";
 
 const { CONTAINER, HOST } = Networking.type;
 
-const getVirtualNetworks = () =>
-  VirtualNetworksStore.getOverlays()
-    .filter(overlay => overlay.enabled && !overlay.subnet6)
-    .map(({ name }) => (
-      <Trans
-        id="Virtual Network: {0}"
-        key={name}
-        render={<option value={`${CONTAINER}.${name}`} />}
-        values={{ 0: name }}
-      />
-    ));
-
-class MultiContainerNetworkingFormSection extends mixin(StoreMixin) {
-  constructor(props) {
-    super(props);
-
-    this.store_listeners = [{ name: "virtualNetworks", events: ["success"] }];
-  }
+export default class MultiContainerNetworkingFormSection extends React.Component<{
+  data: {
+    containers?: Array<{ name: string }>;
+    id: string;
+    networks?: Array<{ mode?: string; name?: string }>;
+    portsAutoAssign: boolean;
+  };
+  errors: object;
+  onAddItem: () => void;
+  onRemoveItem: () => void;
+}> {
+  static configReducers = { networks };
+
+  static defaultProps = {
+    data: {},
+    errors: {},
+    onAddItem() {},
+    onRemoveItem() {}
+  };
 
-  onVirtualNetworksStoreSuccess = () => {
-    this.forceUpdate();
+  state: {
+    virtualNetworks: Overlay[];
+  } = {
+    virtualNetworks: []
   };
 
+  componentDidMount() {
+    VirtualNetworksActions.fetch(virtualNetworks => {
+      this.setState({ virtualNetworks });
```

Comment:
Couldn't we do the `getVirtualNetworks` part here as well? I don't know if this would add performance but it would decrease the confusion that `this.state.virtualNeworks` is only used in the `getVirtualNetworks` function. Or am I missing something?

Reply:
that's a great remark.

`setState` explicitly tells us that the component will rerender with a new internal state. I prefer to only put data into state (and not the `Trans`-components that would be returned by `getVirtualNetworks`) because the type of the state will be more intention-revealing/documenting.

we'll likely have under 10000 options (it should actually be under 10 most of the time, right?) in that select - so i think performance is not something to be concerned about.

local state is used for a single rendering-purpose in a lot of components in dcos-ui. do you think this is confusing and/or should be considered a bad practice?